pkg/apis/migration/v1alpha1/migcluster_types.go

```diff
@@ -498,3 +502,89 @@ func (r *MigCluster) UpdateProvider(provider pvdr.Provider) {
 		}
 	}
 }
+
+func (m *MigCluster) GetInternalRegistryPath(c k8sclient.Client) (string, error) {
+	client, err := m.GetClient(c)
+	if err != nil {
+		return "", err
+	}
+	isList := imgapi.ImageStreamList{}
+	err = client.List(
+		context.TODO(),
+		k8sclient.InNamespace("openshift"),
+		&isList)
+	if err == nil && len(isList.Items) > 0 {
+		if value := isList.Items[0].Status.DockerImageRepository; len(value) > 0 {
+			ref, err := reference.Parse(value)
+			if err == nil {
+				return ref.Registry, nil
+			}
+		}
+	}
+	if client.MajorVersion() != 1 {
+		return "", errors.New(fmt.Sprintf("server version %v.%v not supported. Must be 1.x", client.MajorVersion(), client.MinorVersion()))
+	}
+	if client.MinorVersion() < 7 {
+		return "", errors.New(fmt.Sprintf("Kubernetes version 1.%v not supported. Must be 1.7 or greater", client.MinorVersion()))
+	} else if client.MinorVersion() <= 11 {
+		registrySvc := kapi.Service{}
+		err := client.Get(
+			context.TODO(),
+			k8sclient.ObjectKey{
+				Namespace: "default",
+				Name:      "docker-registry",
+			},
+			&registrySvc)
+		if err != nil {
+			// Return empty registry host but no error; registry not found
+			return "", nil
+		}
+		internalRegistry := registrySvc.Spec.ClusterIP + ":" + strconv.Itoa(int(registrySvc.Spec.Ports[0].Port))
+		return internalRegistry, nil
+	} else {
+		config := kapi.ConfigMap{}
+		err := client.Get(
+			context.TODO(),
+			k8sclient.ObjectKey{
+				Namespace: "openshift-apiserver",
+				Name:      "config",
+			},
+			&config)
+		if err != nil {
+			return "", err
+		}
+		serverConfig := apiServerConfig{}
+		err = json.Unmarshal([]byte(config.Data["config.yaml"]), &serverConfig)
+		if err != nil {
+			return "", err
+		}
+		internalRegistry := serverConfig.ImagePolicyConfig.InternalRegistryHostname
+		if len(internalRegistry) == 0 {
+			return "", nil
+		}
+		return internalRegistry, nil
+	}
+}
+
+func (m *MigCluster) GetRegistryPath(c k8sclient.Client) (string, error) {
+	if len(m.Spec.ExposedRegistryPath) > 0 {
+		return m.Spec.ExposedRegistryPath, nil
+	} else if !m.Spec.IsHostCluster {
+		// not host cluster and no path specified, return empty path
+		return "", nil
+	}
+	return m.GetInternalRegistryPath(c)
+}
+
+type routingConfig struct {
+	Subdomain string `json:"subdomain"`
+}
+type imagePolicyConfig struct {
+	InternalRegistryHostname string `json:"internalRegistryHostname"`
+}
+
+// apiServerConfig stores configuration information about the current cluster
+type apiServerConfig struct {
+	ImagePolicyConfig imagePolicyConfig `json:"imagePolicyConfig"`
+	RoutingConfig     routingConfig     `json:"routingConfig"`
```

Comment:
```
$ oc get openshiftapiservers.operator.openshift.io -oyaml | grep routing -C 2
            f:imagePolicyConfig: {}
            f:projectConfig: {}
            f:routingConfig: {}
            f:servingInfo: {}
            f:storageConfig: {}
--
      projectConfig:
        projectRequestMessage: ""
      routingConfig:
        subdomain: apps.cluster-alpatel-aux-tools-444.alpatel-aux-tools-444.mg.dog8code.com
      servingInfo:
[alpatel@alpatel migsamples]$
```

Reply:
Same as above, I think this might be a better place to get the info

Reply:
Same question about if using the CR to get this info forces us to run on OCP 4. Maybe we are at a point where we should drop OCP 3 support for running controller but want to make sure we are consciously making that decision.

Reply:
So I don't think we're dropping OCP3 support yet since we're only executing this if we're on OCP4, unless I'm missing something.